src/components/FriendRequests.tsx

```diff
@@ -0,0 +1,70 @@
+'use client'
+
+import {FC, useState} from "react";
+import {Check, UserPlus, X} from 'lucide-react';
+import axios from "axios";
+import { useRouter } from 'next/navigation';
+
+interface FriendRequestsProps {
+    incomingFriendRequests: {
```

Comment:
{senderId:string, senderEmail:string} is used more than once. Either declare an interface or find the pre-existing type and use it here.